mlflow/transformers/flavor_config.py

```diff
@@ -0,0 +1,145 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Any, Dict
+
+from mlflow.transformers.hub_utis import get_latest_commit_for_repo
+from mlflow.transformers.torch_utils import _extract_torch_dtype_if_set
+
+if TYPE_CHECKING:
+    import transformers
+
+
+# Flavor configuration keys
+class FlavorKey:
```

Comment:
A random idea. Can we define a class representing a config using `dataclass` so we can access keys as attributes?

```python
# we can do:
conf.xxx # instead of conf['xxx']
```

Reply:
Yes I tried that indeed, but it doesn't seem to be an optimal use case for dataclass due to many dynamic key access like "%s_type".... We need to define all component keys which looks a bit redundant.
```
@dataclass
class FlavorConfig:
   ...
   tokenizer_name: str = None
   tokenizer_type: str = None
   tokenizer_revision: str = None
   image_processor_name: str = None
   image_process_type: str = None
   # repeat for all component types
   ...
```